### PR TITLE
Adds the ability to run a server in the background

### PR DESCRIPTION
**Pull request by desertaxle**
Adds a `--background` (or `-b`) flag to `prefect server start` to allow running a server in a background process. A server started with `--background` can be stopped with `prefect server stop`.

When starting a background server, a `server.pid` file is written to the `PREFECT_HOME` directory with the process ID of the background server. This lets `prefect server stop` know which process to interrupt. Currently, only one background server can be started at a time. Allowing more simultaneous background servers will likely require an update to `prefect server stop`.

Closes https://github.com/PrefectHQ/prefect/issues/6989

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: src/prefect/cli/server.py [hunk 1]
```diff
@@ -153,11 +184,32 @@ async def start(
...
+async def stop():
+    """Stop a Prefect server instance running in the background"""
+    pid_file = anyio.Path(PREFECT_HOME.value() / "server.pid")
+    if not await pid_file.exists():
+        exit_with_error("No server running in the background.")
+    pid = int(await pid_file.read_text())
+    try:
+        os.kill(pid, 15)
+    except ProcessLookupError:
+        exit_with_error("No server running in the background.")
```
Comment (chrisguidry) [inline src/prefect/cli/server.py:210]: I wouldn't consider this case an error.  They wanted the process to not be running, and it's not running.  I'd make this a successful exit (even if you still print the message that no server was running), which should make it easier to use in automated scripts.  (I _would_ keep the message on L205 an error exit code)

File: src/prefect/cli/server.py [hunk 2]
```diff
@@ -116,35 +123,59 @@ async def start(
...
+            exit_with_error(
+                "A server is already running in the background. To stop it,"
+                " run `prefect server stop`."
+            )
+
+    app.console.print(generate_welcome_blurb(base_url, ui_enabled=ui))
+    app.console.print("\n")
+
     try:
-        async with anyio.create_task_group() as tg:
```
Comment (chrisguidry) [inline src/prefect/cli/server.py:120]: :fire:

**Review by chrisguidry — APPROVED** [review]

File: src/prefect/cli/server.py [hunk 3]
```diff
@@ -153,11 +184,34 @@ async def start(
...
+
+    app.console.print("Server stopped!")
+
+
+@server_app.command()
+async def stop():
+    """Stop a Prefect server instance running in the background"""
+    pid_file = anyio.Path(PREFECT_HOME.value() / "server.pid")
+    if not await pid_file.exists():
+        exit_with_error("No server running in the background.")
```
Comment (cicdw) [inline src/prefect/cli/server.py:205]: i feel like this should still be `exit_with_success` - this command feels like it "always succeeds" to me

File: src/prefect/cli/server.py [hunk 4]
```diff
@@ -103,8 +105,13 @@ async def start(
     ),
     late_runs: bool = SettingsOption(PREFECT_API_SERVICES_LATE_RUNS_ENABLED),
     ui: bool = SettingsOption(PREFECT_UI_ENABLED),
+    background: bool = typer.Option(
```
[the remark this answers is not in the document]
Comment (cicdw) [reply src/prefect/cli/server.py:110]: Nah I like `background`; `detach` implies `attach`

**Review by cicdw — APPROVED** [review]
Awesome

**Pull request merged**